src/gsy_e/models/area/scm_manager.py

```diff
@@ -0,0 +1,249 @@
+    community_uuid: str
+    consumption_kWh: float = 0.
+    production_kWh: float = 0.
+    self_consumed_energy_kWh: float = 0.
+    energy_surplus_kWh: float = 0.
+    energy_need_kWh: float = 0.
+    energy_bought_from_community_kWh: float = 0.
+    energy_sold_to_grid_kWh: float = 0.
+
+
+@dataclass
+class HomeEnergyBills:
+    """Represents home energy bills."""
+    base_energy_bill: float
+    gsy_energy_bill: float
+    savings_euro: float
+    savings_percent: float
+
+
+class SCMManager:
+    """Handle the community manager coefficient trade."""
+    def __init__(self, community_uuid: str, time_slot: DateTime):
+        self._home_data: Dict[str, HomeAfterMeterData] = {}
+        self.community_data = CommunityData(community_uuid)
+        self._time_slot = time_slot
+        self._bills: Dict[str, HomeEnergyBills] = {}
+        self._grid_fees_reduction = 0.28
+
+    def add_home_data(self, home_uuid: str, home_name: str,
+                      grid_fees_eur: float, coefficient_percent: float,
+                      market_maker_rate_eur: float, feed_in_tariff_eur: float,
+                      production_kWh: float, consumption_kWh: float):
+        # pylint: disable=too-many-arguments
+        """Import data for one individual home."""
+        if grid_fees_eur is None:
+            grid_fees_eur = 0.0
+        self._home_data[home_uuid] = HomeAfterMeterData(
+            home_uuid, home_name,
+            grid_fees_eur=grid_fees_eur,
+            sharing_coefficient_percent=coefficient_percent,
+            market_maker_rate_eur=market_maker_rate_eur,
+            feed_in_tariff_eur=feed_in_tariff_eur,
+            production_kWh=production_kWh, consumption_kWh=consumption_kWh)
+
+    def calculate_community_after_meter_data(self):
+        """Calculate community data by aggregating all single home data."""
+        self.community_data.production_kWh = sum(
+            data.production_kWh for data in self._home_data.values())
+        self.community_data.consumption_kWh = sum(
+            data.consumption_kWh for data in self._home_data.values())
+        self.community_data.self_consumed_energy_kWh = sum(
+            data.self_consumed_energy_kWh for data in self._home_data.values())
+        self.community_data.energy_surplus_kWh = sum(
+            data.energy_surplus_kWh for data in self._home_data.values())
+        self.community_data.energy_need_kWh = sum(
+            data.energy_need_kWh for data in self._home_data.values())
+
+        for home_data in self._home_data.values():
+            home_data.set_community_production(self.community_data.energy_surplus_kWh)
+
+        self.community_data.energy_bought_from_community_kWh = sum(
+            data.energy_bought_from_community_kWh for data in self._home_data.values())
+        self.community_data.energy_sold_to_grid_kWh = sum(
+            data.energy_sold_to_grid_kWh for data in self._home_data.values())
+
+    def calculate_home_energy_bills(
```

Comment:
I will look more closely into this central method tomorrow, but I have the feeling, that it could be divided into multiple methods, also to make it more readable. DO you agree?

Reply:
I have already done some fixes in this method for my next task in this epic (I have essentially moved large parts of it in separate methods of the HomeEnergyBills class as part of fully calculating the energy bills) therefore if you can tolerate this to be dealt with at the results PR it would be great! If you think though that it is cleaner to do it now, then let me know and I'll change it here, no problem at all! Thanks in advance

Reply:
Ah no. I can wait for the results PR.